guides/rag-chatbot-offline-evaluation.mdx

```diff
@@ -0,0 +1,391 @@
+---
+title: Build and evaluate a RAG chatbot with Chainlit, OpenAI, Pinecone and Literal
+---
+
+In this guide you will learn how to build, monitor, evaluate and improve a conversational application.
+We will use [Chainlit](https://docs.chainlit.io/get-started/overview) in combination with Literal, 
+to show the development flow of building, evaluating and improving the chatbot. The chatbot uses Retrieval
+Augmented Generation (RAG) to retrieve contexts for answer generation. 
+
+## 1. Build a RAG chatbot with Chainlit
+
+First, we pick Pinecone as vector database within our RAG. Another vector store could also be used.
+Make sure to store your API keys of Literal, OpenAI and Pinecone in a `.env` file as follows.
+Make sure you have data documents stored in this vector database.
+
+<CodeGroup>
+```bash .env
+LITERAL_API_KEY=
+OPENAI_API_KEY=
+PINECONE_API_KEY=
+PINECONE_CLOUD=
+PINECONE_REGION=
+```
+
+```bash requirements.txt
+openai>=1.9.0
+pinecone-client>=3.0.2
+chainlit>=1.0.502
+literalai>=0.0.501
+ragas>=0.1.7
+``` 
+</CodeGroup>
+
+The whole code for the chatbot lives in a single file `app.py`, which you can browse
+at [Documentation - RAG application](https://github.com/Chainlit/Documentation-RAG-application).
+
+Beyond the initial imports, here is how the Python code for this chatbot is structured. 
+1. First, we create a Pinecone vector index. This vector database is used to store the documents and the
+embeddings of the data that you want your chatbot to use. For example, you can embed and store pages
+of a technical product documentation.  <br/><br/>
+Check out the [Embed the documentation](https://github.com/Chainlit/Documentation-RAG-application#embed-the-documentation) 
+section to populate your Pinecone index with your own documents.
+
+<CodeGroup>
+```python Set up Pinecone
+pinecone_client = Pinecone(api_key=os.environ.get("PINECONE_API_KEY"))
+pinecone_spec = ServerlessSpec(
+    cloud=os.environ.get("PINECONE_CLOUD"),
+    region=os.environ.get("PINECONE_REGION"),
+)
+
+def create_pinecone_index(name, client, spec):
+    """
+    Create a Pinecone index if it does not already exist.
+    """
+    if name not in client.list_indexes().names():
+        client.create_index(name, dimension=1536, metric="cosine", spec=spec)
+    return pinecone_client.Index(name)
+
+pinecone_index = create_pinecone_index(
+    "literal-rag-index", pinecone_client, pinecone_spec
+)
+```
+</CodeGroup>
+
+
+2. We then use a prompt defined from a file - or fetched from Literal, and initialize
+a user session with it. When a user question comes in, we trigger our RAG agent logic, 
+which we will study in the next step.
+
+<CodeGroup>
+```python Setup chat prompt template
+prompt_path = os.path.join(os.getcwd(), "app/prompts/rag.json")
```

Comment:
Give an example of what is in this json file, otherwise people can't work with this.